source/libs/fetch-dom.ts

```diff
@@ -0,0 +1,11 @@
+import domify from 'doma';
+
+async function fetchDom(url: string): Promise<DocumentFragment>;
+async function fetchDom(url: string, selector: string): Promise<Element>;
+async function fetchDom(url: string, selector?: string): Promise<Node> {
+	const response = await fetch(url);
+	const dom = domify(await response.text());
+	return selector ? dom.querySelector(selector) : dom;
+}
+
+export default fetchDom;
```

Comment:
This can be reused in other parts of RGH too. `doma` is almost only used for this purpose in RGH.